parser-java/api/src/main/java/org/jboss/forge/addon/parser/java/ui/AbstractJavaSourceCommand.java

```diff
@@ -63,13 +59,20 @@
          @FacetConstraint(value = JavaSourceFacet.class, type = FacetConstraintType.REQUIRED)
 })
 public abstract class AbstractJavaSourceCommand<SOURCETYPE extends JavaSource<?>> extends AbstractProjectCommand
+      implements JavaSourceDecorator<SOURCETYPE>
 {
    private UIInput<String> targetPackage;
    private UIInput<String> named;
    private UIInput<Boolean> overwrite;
    private UIInput<String> extendsType;
    private UIInputMany<String> implementsType;
 
+   private JavaSourceDecorator<SOURCETYPE> delegate = this;
+
+   public void setJavaSourceDecorator(JavaSourceDecorator<SOURCETYPE> delegate) {
```

Comment:
Regarding passing `this`, I used to have it in the parameters but that complicates things. For example, in the `RestNewEndpointCommand` case, `this` at the calling site would be an `AbstractJavaSourceCommand` but the decorator itself needs it to be a `RestNewEndpointCommand`, which would force users to downcast… That said, I already need to downcast when I instantiate the decorator in my code (I pass `this` to my decorator constructor) but, at least, I don't need to deal with it anymore in the decorator itself.